ASP.NET API/WAVC_WebApi/Controllers/ApplicationUserController.cs

```diff
@@ -1,124 +1,88 @@
+        UserManager<ApplicationUser> userManager;
+        ApplicationDbContext dBContext;
+        FriendsManager friendsManager;
+        int searchResults = 10;
 
-        public ApplicationUserController(
-            UserManager<ApplicationUser> userManager, SignInManager<ApplicationUser> signInManager,
-            ApplicationDbContext context, IOptions<ApplicationSettings> appSettings)
+        public ApplicationUserController(UserManager<ApplicationUser> userManager, ApplicationDbContext dBContext)
         {
-            _userManager = userManager;
-            _signInManager = signInManager;
-            _dbContext = context;
-            _applicationSettings = appSettings.Value;
+            this.userManager = userManager;
+            this.dBContext = dBContext;
+            friendsManager = new FriendsManager(dBContext);
         }
 
-        [HttpPost]
-        [Route("Register")]
-        //POST : /api/ApplicationUser/Register
-        public async Task<Object> PostApplicationUserAsync(ApplicationUserModel model)
+        [HttpGet]
+        [Route("[action]")]
+        public async Task<List<ApplicationUserModel>> GetFriends()
         {
-            var applicationUser = new ApplicationUser()
+            var user = await userManager.GetUserAsync(HttpContext.User);
+            if (user != null)
             {
-                FirstName = model.FirstName,
-                LastName = model.LastName,
-                PhoneNumber = model.PhoneNumber,
-                Email = model.Email,
-                UserName = model.UserName
-            };
-
-            try
-            {
-                var result = await _userManager.CreateAsync(applicationUser, model.Password);
-                return Ok(result);
+                return friendsManager.GetFriends(user).Select(u => u.ToApplicationUserModel()).ToList();
             }
-            catch (Exception)
+            else
             {
-                throw;
+                return new List<ApplicationUserModel>();
             }
         }
+
         [HttpGet]
-        //GET : /api/
-        public  List<ApplicationUser> GetUsers()
+        [Route("[action]")]
+        public async Task<List<ApplicationUserModel>> GetRequestsForUser()
         {
-            return  _userManager.Users.ToList();
+            var user = await userManager.GetUserAsync(HttpContext.User);
+            if (user != null)
+            {
+                return friendsManager.GetRequestsForUser(user).Select(u => u.ToApplicationUserModel()).ToList();
+            }
+            else
+            {
+                return new List<ApplicationUserModel>();
+            }
         }
 
-        [HttpGet("{id}/Friends")]
-        //GET : /api/5/friends
-        public async Task<ActionResult<ApplicationUser>> GetFriendsAsync(string id)
+        [HttpGet]
+        [Route("[action]")]
+        public async Task<ApplicationUserModel> GetCurrentFriend(string name, string surname)
```

Comment:
TBH. I didn't think about which English I use. Surname seemed more natural to me and that may be because we were tought British English in school. As for which one makes more sense, I'd say that in some countries people write their familly name first and their name second, so First Name and Last Name don't make much sense in such situation.